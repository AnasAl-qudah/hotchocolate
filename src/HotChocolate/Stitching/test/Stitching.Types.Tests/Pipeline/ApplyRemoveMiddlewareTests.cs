using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;
using HotChocolate.Language;
using HotChocolate.Stitching.Types.Pipeline.ApplyRemove;
using HotChocolate.Stitching.Types.Pipeline.PrepareDocuments;
using Snapshooter.Xunit;
using Xunit;

namespace HotChocolate.Stitching.Types.Pipeline;

public class ApplyRemoveMiddlewareTests
{
    [Fact]
    public async Task Apply_TypeRemove()
    {
        // arrange
        MergeSchema pipeline = CreatePipeline();

        var service = new ServiceConfiguration(
            "abc",
            documents: new[]
            {
                Utf8GraphQLParser.Parse(@"
                type Foo {
                    abc: String
                }"),
                Utf8GraphQLParser.Parse(@"
                extend type Foo @remove")
            });

        var configurations = new List<ServiceConfiguration> { service };
        var context = new SchemaMergeContext(configurations);

        // act
        await pipeline.Invoke(context);

        // assert
        context.Documents.Single().SyntaxTree.ToString().MatchSnapshot();
    }

    [Fact]
    public async Task Apply_FieldRemove()
    {
        // arrange
        MergeSchema pipeline = CreatePipeline();

        var service = new ServiceConfiguration(
            "abc",
            documents: new[]
            {
                Utf8GraphQLParser.Parse(@"
                type Foo {
                    abc: String
                }"),
                Utf8GraphQLParser.Parse(@"
                extend type Foo {
                    abc: String @remove
                }")
            });

        var configurations = new List<ServiceConfiguration> { service };
        var context = new SchemaMergeContext(configurations);

        // act
        await pipeline.Invoke(context);

        // assert
        context.Documents.Single().SyntaxTree.ToString().MatchSnapshot();
    }

    private MergeSchema CreatePipeline()
        => new SchemaMergePipelineBuilder()
            .Use(next =>
            {
                var middleware = new PrepareDocumentsMiddleware(next);
                return context => middleware.InvokeAsync(context);
            })
            .Use(next =>
            {
                var middleware = new ApplyRemoveMiddleware(next);
                return context => middleware.InvokeAsync(context);
            })
            .Compile();
}
