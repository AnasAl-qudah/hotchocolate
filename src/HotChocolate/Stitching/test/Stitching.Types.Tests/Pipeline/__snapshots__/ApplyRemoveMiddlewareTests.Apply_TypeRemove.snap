extend schema @_hc_service(name: "abc") {

}
